python/ray/worker.py

```diff
@@ -846,21 +863,26 @@ def _process_task(self, task, function_execution_info):
             self._handle_process_task_failure(
                 function_descriptor, return_object_ids, e, traceback_str)
             return
+        finally:
+            self._return_object_ids = []
 
         # Store the outputs in the local object store.
-        try:
-            with profiling.profile("task:store_outputs", worker=self):
-                # If this is an actor task, then the last object ID returned by
-                # the task is a dummy output, not returned by the function
-                # itself. Decrement to get the correct number of return values.
-                num_returns = len(return_object_ids)
-                if num_returns == 1:
-                    outputs = (outputs, )
-                self._store_outputs_in_object_store(return_object_ids, outputs)
-        except Exception as e:
-            self._handle_process_task_failure(
-                function_descriptor, return_object_ids, e,
-                ray.utils.format_error_message(traceback.format_exc()))
+        if not isinstance(outputs, RayNoReturn):
```

Comment:
This check should be inside of `_store_outputs_in_object_store` instead of here. Also, `outputs` is a list of outputs and so we need to do the check on each element of that list.